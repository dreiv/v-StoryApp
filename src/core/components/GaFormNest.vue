<template>
  <slot />

  <div v-if="$slots.nested" :class="$style.nested">
    <slot name="nested" />
  </div>
</template>

<style module>
.nested {
  display: flex;
  flex-direction: column;
  gap: var(--ga-size-spacing-03);

  margin-left: var(--ga-size-spacing-06);
}
</style>
