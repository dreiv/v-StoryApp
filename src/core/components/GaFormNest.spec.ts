import { shallowMount, type VueWrapper } from '@vue/test-utils'
import GaFormNest from './GaFormNest.vue'

describe('GaFormNest', () => {
  let wrapper: VueWrapper

  it('should render correctly with only the default slot', () => {
    wrapper = shallowMount(GaFormNest, { slots: { default: 'Default Slot Content' } })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render correctly with both default and nested slots', () => {
    wrapper = shallowMount(GaFormNest, {
      slots: {
        default: 'Default Slot Content',
        nested: 'Nested Slot Content',
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
