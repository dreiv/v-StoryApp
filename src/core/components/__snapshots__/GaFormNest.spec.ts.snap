// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GaFormNest > should render correctly with both default and nested slots 1`] = `
"Default Slot Content
<div class="nested">Nested Slot Content</div>"
`;

exports[`GaFormNest > should render correctly with only the default slot 1`] = `
"Default Slot Content
<!--v-if-->"
`;
